test/cppsuite/test_harness/core/configuration.h

```diff
@@ -112,99 +75,19 @@ class configuration {
-    }
-
-    bool
-    get_bool(const std::string &key)
-    {
-        return get<bool>(key, false, types::BOOL, false, config_item_to_bool);
-    }
-
-    bool
-    get_optional_bool(const std::string &key, const bool def)
-    {
-        return get<bool>(key, true, types::BOOL, def, config_item_to_bool);
-    }
-
-    int64_t
-    get_int(const std::string &key)
-    {
-        return get<int64_t>(key, false, types::INT, 0, config_item_to_int);
-    }
-
-    int64_t
-    get_optional_int(const std::string &key, const int64_t def)
-    {
-        return get<int64_t>(key, true, types::INT, def, config_item_to_int);
-    }
-
-    configuration *
-    get_subconfig(const std::string &key)
-    {
-        return get<configuration *>(key, false, types::STRUCT, nullptr,
-          [](WT_CONFIG_ITEM item) { return new configuration(item); });
-    }
-
-    configuration *
-    get_optional_subconfig(const std::string &key)
-    {
-        return get<configuration *>(key, true, types::STRUCT, nullptr,
-          [](WT_CONFIG_ITEM item) { return new configuration(item); });
-    }
-
-    std::vector<std::string>
-    get_list(const std::string &key)
-    {
-        return get<std::vector<std::string>>(key, false, types::LIST, {}, config_item_to_list);
-    }
-
-    std::vector<std::string>
-    get_optional_list(const std::string &key)
-    {
-        return get<std::vector<std::string>>(key, true, types::LIST, {}, config_item_to_list);
-    }
+    std::string get_string(const std::string &key);
+    std::string get_optional_string(const std::string &key, const std::string &def);
+    bool get_bool(const std::string &key);
+    bool get_optional_bool(const std::string &key, const bool def);
+    int64_t get_int(const std::string &key);
+    int64_t get_optional_int(const std::string &key, const int64_t def);
+    configuration *get_subconfig(const std::string &key);
+    configuration *get_optional_subconfig(const std::string &key);
+    std::vector<std::string> get_list(const std::string &key);
+    std::vector<std::string> get_optional_list(const std::string &key);
 
     private:
-    static bool
-    config_item_to_bool(const WT_CONFIG_ITEM item)
-    {
-        return (item.val != 0);
-    }
-
-    static int64_t
-    config_item_to_int(const WT_CONFIG_ITEM item)
-    {
-        return (item.val);
-    }
-
-    static std::string
-    config_item_to_string(const WT_CONFIG_ITEM item)
-    {
-        return std::string(item.str, item.len);
-    }
-
-    static std::vector<std::string>
-    config_item_to_list(const WT_CONFIG_ITEM item)
-    {
-        auto str = config_item_to_string(item);
-
-        /* Get rid of the brackets. */
-        testutil_assert(!str.empty() && str.front() == '[' && str.back() == ']');
-        str.pop_back();
-        str.erase(0, 1);
-
-        return (split_string(str, ','));
-    }
+    enum class types { BOOL, INT, LIST, STRING, STRUCT };
```

Comment:
I see that this got moved into the private section, I guess that's fine given nothing else uses it.

Reply:
Yes that enum is only used by private methods of the the class.